src/compiler/scala/tools/nsc/settings/StandardScalaSettings.scala

```diff
@@ -52,7 +51,26 @@ trait StandardScalaSettings { _: MutableSettings =>
   val nowarn =         BooleanSetting ("-nowarn", "Generate no warnings.") withAbbreviation "--no-warnings" withPostSetHook { s => if (s) maxwarns.value = 0 }
   val optimise:        BooleanSetting // depends on post hook which mutates other settings
   val print =          BooleanSetting ("-print", "Print program with Scala-specific features removed.") withAbbreviation "--print"
-  val target =         ChoiceSetting  ("-target", "target", "Target platform for object files.", AllTargetVersions, "8") withPreSetHook normalizeTarget _ withAbbreviation "--target"
+  val release =
+    ChoiceSetting("-release", "release", "Compile for a version of the Java API and target class file.", AllTargetVersions, normalizeTarget(javaSpecVersion))
+      .withPostSetHook { setting =>
+        val current = setting.value.toInt
+        if (!isJavaAtLeast("9") && current > 8) errorFn.apply("-release is only supported on JVM 9 and higher")
+        if (target.valueSetByUser.map(_.toInt > current).getOrElse(false)) errorFn("-release cannot be less than -target")
+        //target.value = setting.value  // this would trigger deprecation
+      }
```

Comment:
There seems to be no check here preventing one from using e.g. `-release 17` when running the compiler on JDK 11

Reply:
I think it can just build the "choices" correctly.

There's also a compatibility problem, current is `-release 8`, but choice accepts only `-release:8` for some reason. I think it is OK to fix choice processing to accept either form, which I will try.

Reply:
Also thanks, now I see I forgot
`// TODO validate release <= java.specification.version`